### PR TITLE
Add queryHash to query info and metadata

### PR DESCRIPTION
**Pull request by LeonSo7**
This is needed for additional logging of queryHash for query completion events.

**Test plan:**
The following unit tests were updated:
  - `TestQueryInfo`
  -  `TestBasicQueryInfo` 

== NO RELEASE NOTE ==

Depended by https://github.com/facebookexternal/presto-facebook/pull/1928

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import java.math.BigInteger;
+
+import static com.google.common.hash.Hashing.sha256;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class QueryInfoUtils
+{
+    private QueryInfoUtils() {}
+
+    public static String convertToQueryHash(String query)
```
Comment (prithvip) [inline presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java:28]: Rename to computeQueryHash

File: presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        HashFunction messageDigest = sha256();
```
Comment (prithvip) [inline presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java:30]: A cryptographic hash function is unnecessary here. Let's use something fast like xxhash64 instead.

File: presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        byte[] queryBytes = query.replaceAll("\\s+", " ").trim().toUpperCase().getBytes(UTF_8);
```
Comment (prithvip) [inline presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java:31]: Would it be more accurate to go through the sql formatter first?

**Review by prithvip — CHANGES_REQUESTED** [review]
Let's also add a few tests showing that the hash isn't affected by whitespace

File: presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java (hunk 1 above)
Comment (LeonSo7) [reply presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java]: > Let's also add a few tests showing that the hash isn't affected by whitespace

As discussed offline, we go through `sqlFormatter` upstream. Query string should be consistently whitespace formatted.

File: presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java (hunk 3 above)
Comment (LeonSo7) [reply presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java]: As discussed offline, we go through `sqlFormatter` upstream. Query string should be consistently whitespace formatted.

File: presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java (hunk 2 above)
Comment (LeonSo7) [reply presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java]: Updated to use xxhash64.

File: presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    private QueryInfoUtils() {}
+
+    public static String computeQueryHash(String query)
+    {
+        if (query.isEmpty()) {
+            return "";
+        }
+
+        byte[] queryBytes = query.getBytes(UTF_8);
+        long queryHash = XxHash64.hash(new BigInteger(queryBytes).longValue());
```
Comment (LeonSo7) [inline presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java:33]: XxHash64's hash function does not accept `String` types as input. We need to convert the bytes to a long. To do this, we use the `BigInteger` class which has a `longValue() method` commonly used to convert a byte array to a long value.

File: presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+public class QueryInfoUtils
+{
+    private QueryInfoUtils() {}
+
+    public static String convertToQueryHash(String query)
+    {
+        HashFunction messageDigest = sha256();
+        byte[] queryBytes = query.replaceAll("\\s+", " ").trim().toUpperCase().getBytes(UTF_8);
+        HashCode queryHashCode = messageDigest.hashBytes(queryBytes);
+        BigInteger hashCodeBigIntegerValue = new BigInteger(1, queryHashCode.asBytes());
```
[the remark this answers is not in the document]
Comment (LeonSo7) [reply presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java]: As below comment:
> As discussed offline, we use `BigInteger` to convert to a `long`. `XxHash64`'s hash function accepts `long` as input.

File: presto-main/src/test/java/com/facebook/presto/util/TestQueryInfoUtils.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+package com.facebook.presto.util;
+
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.util.QueryInfoUtils.computeQueryHash;
+import static com.google.common.base.Strings.repeat;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestQueryInfoUtils
```
[the remark this answers is not in the document]
Comment (LeonSo7) [reply presto-main/src/test/java/com/facebook/presto/util/TestQueryInfoUtils.java:23]: Updated to add this test.

File: presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    private QueryInfoUtils() {}
+
+    public static String computeQueryHash(String query)
+    {
+        if (query.isEmpty()) {
+            return "";
+        }
+
+        byte[] queryBytes = query.getBytes(UTF_8);
+        long queryHash = hash(new BigInteger(queryBytes).longValue());
```
[the remark this answers is not in the document]
Comment (LeonSo7) [reply presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java]: Updated to use the `update(byte[] data)` method.

**Review by prithvip — APPROVED** [review]
LGTM

**Review by swapsmagic — APPROVED** [review]
LGTM

File: presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+import io.airlift.slice.XxHash64;
+
+import static java.lang.Long.toHexString;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class QueryInfoUtils
+{
+    private QueryInfoUtils() {}
+
+    public static String computeQueryHash(String query)
```
Comment (tdcmeehan) [inline presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java:25]: Nit: please add a preconditon.

`requireNonNull(query);`

**Review by tdcmeehan — APPROVED** [review]
LGTM % one optional not.  Let me know if you're going to get to that, otherwise happy to merge.

File: presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java (hunk 8 above)
Comment (LeonSo7) [reply presto-main/src/main/java/com/facebook/presto/util/QueryInfoUtils.java:26]: Updated to add the precondition :).

**Pull request merged**